src/components/DynamicContacts/DynamicContacts.js

```diff
@@ -0,0 +1,61 @@
+import React, {useState} from 'react';
+import Contacts from '../Contacts/Contacts';
+
+function DynamicContacts (data) {
+    const fiveContacts = data.data.slice(0,5)
+
+    let [contact, setDOM] = useState(fiveContacts)
+    console.log("contact",contact)
+
+    //Funciona bien
+    const handleDelete = contactId => { 
+        const deletedContact = contact.filter(contact => contact.id !== contactId)
+        setDOM( deletedContact )
+    }
+    //Funciona bien
+    const addRandomContact = () => {
+        console.log("add")
+            let randomNum = Math.floor(Math.random() * data.data.length)
+            let randomContact = data.data[randomNum]
+            setDOM(state => state.concat(randomContact))
+    }
+    //No funciona
+    const sortByName = () => {
+        console.log("sort by name")
+        setDOM(item => item.sort((a, b) =>
+        a.name - b.name
+        ))
+        console.log("sorted by name:",contact)
+    }
+    //Funciona en consola pero no lo pinta en el DOM. Sí ordena cuando añades un nuevo contacto o lo borras... 
+    const sortByPopularity = () => {
+        console.log("sort by popularity")
+        setDOM(item => item.sort((a,b)=>
```

Comment:
`setDom(state => [...state].sort((a,b) => b.popularity - a.popularity)`

Reply:
React compara estados com Object.is (basicamente).
Esto quiere decir que compara referencias.
Cuando hablamos de referencia nos referimos a esto:

```
const a = {}
const b = {}
a === b -> false
{} === {} -> false

const c = {}
const d = c
d === c -> true
```
Sort a diferencia de map o filter muta el array, no devuelve una copia.
Esto quiere decir que el array que vos estas actualizando contiene la misma referencia que el array anterior (ocupan el mismo espacio en memoria).
React piensa que nada cambio.

En mi ejemplo, con el `[...state]` primero haces una copia del array original y despues haces el sort, la referencia cambia. Esto deberia de funcionar